tests/unit/core/server/testServer.js

```diff
@@ -262,5 +272,31 @@ describe(__filename, () => {
       expect(reduxStoreState.user).toEqual(user);
       mockUserApi.verify();
     });
+
+    it('performs a server redirect when requested by the app', async () => {
+      const { store, sagaMiddleware } = createAmoStore();
+
+      const addon = fakeAddon;
+      mockApi
+        .expects('fetchAddon')
+        .once()
+        .resolves(createAddonsApiResult([addon]));
+
+      const client = testClient({
+        routes: amoRoutes,
+        appSagas: addonsSaga,
+        sagaMiddleware,
+        store,
+      });
+
+      const response = await client
+        .get(`/en-US/firefox/addon/${addon.id}/`)
```

Comment:
I'm not a fan of creating functional tests like this. I know it's nice for regressions but they are always a headache to maintain. This test suite is meant to be a unit test of server functionality (but naturally it integrates a few lower level units).

What I'd like to see instead is a stub route added for a brand new component (just for testing) that dispatches the server redirect action in its `componentWillMount()` code. The test should look more like this: https://github.com/willdurand/addons-frontend/blob/aa085219b8379a9fa24bcd0bc02d6290f3decc4a/tests/unit/core/server/testServer.js#L106

This test will independently verify that any component which dispatches a redirect action will trigger the server redirect. Since we already have a unit test to guarantee that `<Addon>` dispatches this action then we can trust that it will work.

Reply:
I agree. Ultimately, I'd like to see more integration tests like this, or even better, some end-to-end tests.